### ISSUE TITLE
why gpu utility very low

### ISSUE THREAD

**Issue by dulvqingyunLT** [facebookresearch/mmf#1166]
## ❓ Questions and Help
When I finetune model visual-bert on GPU A10(only one gpu), the gpu utility is very low, and i use pytorch profiler founding that 60% of training time is consumed by DataLoader.

the command is: 
mmf_run config=projects/m4c/configs/textvqa/defaults.yaml \
    datasets=vqa2 \
    model=visual_bert \
    run_type=train_val

details of defualts.yaml as follows:

model_config:
  visual_bert:
    hidden_size: 768
    hidden_dropout_prob: 0.1
    training_head_type: classification
    num_labels: 3129
    pooler_strategy: vqa
    losses:
    - type: logit_bce

dataset_config:
  vqa2:
    return_features_info: true
    processors:
      text_processor:
        type: bert_tokenizer
        params:
          tokenizer_config:
            type: bert-base-uncased
            params:
              do_lower_case: true
          mask_probability: 0
          max_seq_length: 128

optimizer:
  type: adam_w
  params:
    lr: 5e-5
    eps: 1e-8

scheduler:
  type: warmup_linear
  params:
    num_warmup_steps: 6000
    num_training_steps: 60000

evaluation:
  metrics:
  - vqa_accuracy

training:
  batch_size: 64
  num_workers: 6
  fast_read: true
  prefetch_factor: 64
  lr_scheduler: true
  pin_memory: true
  # Don't forget to update schedule_attributes if you update this
  max_updates: 60000
  early_stop:
    criteria: vqa2/vqa_accuracy
    minimize: false

checkpoint:
  pretrained_state_mapping:
    model.bert: model.bert

**Comment by apsdehal** [comment]
Try decreasing your num_workers. It might be too many for a single GPU which means most of the time is spent loading the data.

**Comment by dulvqingyunLT** [comment]
> Try decreasing your num_workers. It might be too many for a single GPU which means most of the time is spent loading the data.

 Thanks, I tried to set num_workers to 1, but the problem remained. When I try other model such as visual-bert using the default config, I encounter the same problem.
